lib/js/src/rpc/messages/RegisterAppInterface.js

```diff
@@ -0,0 +1,381 @@
+/*
+* Copyright (c) 2019, Livio, Inc.
+* All rights reserved.
+*
+* Redistribution and use in source and binary forms, with or without
+* modification, are permitted provided that the following conditions are met:
+*
+* Redistributions of source code must retain the above copyright notice, this
+* list of conditions and the following disclaimer.
+*
+* Redistributions in binary form must reproduce the above copyright notice,
+* this list of conditions and the following
+* disclaimer in the documentation and/or other materials provided with the
+* distribution.
+*
+* Neither the name of the Livio Inc. nor the names of its contributors
+* may be used to endorse or promote products derived from this software
+* without specific prior written permission.
+*
+* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+* AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+* IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+* ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+* LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+* CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+* SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+* INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+* CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+* ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+* POSSIBILITY OF SUCH DAMAGE.
+*/
+
+import { RpcRequest } from '../RpcRequest.js';
+import { SdlMsgVersion } from '../structs/SdlMsgVersion.js';
+import { TTSChunk } from '../structs/TtsChunk.js';
+import { DeviceInfo } from '../structs/DeviceInfo.js';
+import { AppInfo } from '../structs/AppInfo.js';
+import { TemplateColorScheme } from '../structs/TemplateColorScheme.js';
+
+import { Language } from '../enums/Language.js';
+import { AppHMIType } from '../enums/AppHMIType.js';
+import { FunctionID } from '../enums/FunctionID.js';
+
+class RegisterAppInterface extends RpcRequest {
+
+    constructor(store) {
+        super(store);
+        this.setFunctionName(FunctionID.REGISTER_APP_INTERFACE);
+    }
+
+    /**
+    * @param {SdlMsgVersion} The max RPC Spec version supported by this library
+    * @return {RegisterAppInterface}
+    */
+    setSdlMsgVersion(sdlMsgVersion) {
+        this.validateType(SdlMsgVersion, sdlMsgVersion);
+
+        this.setParameter(RegisterAppInterface.KEY_SDL_MSG_VERSION, sdlMsgVersion);
+        return this;
+    }
+
+    /**
+    * @return {SdlMsgVersion}
+    */
+    getSdlMsgVersion() {
+        return this.getParameter(RegisterAppInterface.KEY_SDL_MSG_VERSION);
```

Comment:
As I understand there should be `this.getObject(SyncMsgVersion, RegisterAppInterface.KEY_SYNC_MSG_VERSION);` instead of your `getParameter` is it right?

Reply:
See message above